lib/resolvers.js

```diff
@@ -29,3 +30,17 @@ export const mapLinkList = fieldName => doc => {
     return linkItem;
   }).filter(l => l !== undefined);
 };
+
+export const mapScheduleList = fieldName => doc => {
+  const schedule = doc[fieldName];
+
+  if (!schedule) { return []; }
+  return schedule.map((item) => {
```

Comment:
I think we would def need to benchmark here since we're talking about performance, especially if we're just hoping that the runtime is going to optimise for us.